# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: fc5e1bee1332
# Full Hash: fc5e1bee13321946704a915ef4c7c80c0941dcb3
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Send two records with every ObjectStoreCursorResponse. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43252
# ==============================================================================

diff -r 63cf0966cb41 -r fc5e1bee1332 dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Tue Oct 01 09:15:47 2019 +0000
+++ b/dom/indexedDB/IDBObjectStore.cpp	Tue Oct 01 09:15:49 2019 +0000
@@ -1668,6 +1668,8 @@
 
   mTransaction->StartRequest(request, params);
 
+  mTransaction->InvalidateCursorCaches();
+
   return request.forget();
 }
 
@@ -1733,6 +1735,11 @@
         IDB_LOG_STRINGIFY(keyRange), IDB_LOG_STRINGIFY(aLimit));
   }
 
+  // TODO: This is necessary to preserve request ordering only. Proper
+  // sequencing of requests should be done in a more sophisticated manner that
+  // doesn't require invalidating cursor caches (Bug 1580499).
+  mTransaction->InvalidateCursorCaches();
+
   mTransaction->StartRequest(request, params);
 
   return request.forget();
@@ -1958,6 +1965,11 @@
       IDB_LOG_STRINGIFY(mTransaction), IDB_LOG_STRINGIFY(this),
       IDB_LOG_STRINGIFY(keyRange));
 
+  // TODO: This is necessary to preserve request ordering only. Proper
+  // sequencing of requests should be done in a more sophisticated manner that
+  // doesn't require invalidating cursor caches (Bug 1580499).
+  mTransaction->InvalidateCursorCaches();
+
   mTransaction->StartRequest(request, params);
 
   return request.forget();
@@ -2014,6 +2026,8 @@
 
   mTransaction->StartRequest(request, params);
 
+  mTransaction->InvalidateCursorCaches();
+
   return request.forget();
 }
 